src/api/layers.api.js

```diff
@@ -135,46 +136,131 @@ export class WMTSLayer extends Layer {
 }
 
 export class WMSLayer extends Layer {
-    constructor(name, id, opacity, baseURL, format) {
-        super(name, LayerTypes.WMS, id, opacity, baseURL);
+    /**
+     * @param {String} name the name of this layer (lang specific)
+     * @param {String} id the ID of this layer in the BOD
+     * @param {Number} opacity the opacity to apply to this layer
+     * @param {String} baseURL the backend to call for tiles
+     * @param {String} format in which image format the backend must be requested
+     * @param {Number} gutter how much of a gutter we want (specific for tiled WMS, if unset this layer will be a single tile WMS)
+     */
+    constructor(name, id, opacity, baseURL, format, gutter = -1) {
+        super(name, LayerTypes.WMS, id, opacity, false, baseURL);
         this.format = format;
+        this.gutter = gutter;
     }
 
     getURL() {
-        return `${this.baseURL}?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&FORMAT=image%2F${this.format}&TRANSPARENT=true&LAYERS=${this.id}&LANG=en`;
+        return `${this.baseURL ? this.baseURL : WMS_BASE_URL}?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&FORMAT=image%2F${this.format}&TRANSPARENT=true&LAYERS=${this.id}&LANG=en`;
```

Comment:
shouldn't the language be configurable as well? Imho there are (unfortunately) language dependant WMS, am I right @ltclm  / @rebert ?

Reply:
ho right, I thought about it and forgot to implement that.
plus the WMS' URL should change on lang change, so I should look a bit more into that

Reply:
has been implemented